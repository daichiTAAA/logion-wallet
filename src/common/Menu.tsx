import React from 'react';

import MenuItemComponent, { MenuItemData } from './MenuItem';

import './Menu.css';

export interface Props {
    items: MenuItemData[],
}

export default function Menu(props: Props) {

    return (
        <div className="Menu">
            {
                props.items.map(item => (
                    <MenuItemComponent key={ item.id } item={ item } />
                ))
            }
        </div>
    );
}
