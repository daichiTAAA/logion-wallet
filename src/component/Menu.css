.Menu {
    min-height: calc(100% - 160px);
}
