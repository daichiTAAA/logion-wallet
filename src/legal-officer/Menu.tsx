import React from 'react';
import { Link } from 'react-router-dom';

import {
    TOKENIZATION_REQUESTS_PATH,
    PROTECTION_REQUESTS_PATH
} from './LegalOfficerPaths';

export default function Menu() {

    return (
        <ul>
            <li><Link to={ TOKENIZATION_REQUESTS_PATH }>Tokenization Requests</Link></li>
            <li><Link to={ PROTECTION_REQUESTS_PATH }>Protection Requests</Link></li>
        </ul>
    );
}
