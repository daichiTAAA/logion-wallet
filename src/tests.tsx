import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
