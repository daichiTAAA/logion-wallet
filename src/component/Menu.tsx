import React from 'react';

import MenuItemComponent, { MenuItemData } from './MenuItem';
import './Menu.css';

export interface Props {
    items: MenuItemData[]
}

export default function Menu(props: Props) {

    return (
        <div className="Menu">
            {
                props.items.map(item => (
                    <MenuItemComponent item={ item } />
                ))
            }
        </div>
    );
}
