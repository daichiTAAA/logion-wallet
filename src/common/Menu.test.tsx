import { shallowRender } from '../tests';

import Menu from './Menu';

test("renders", () => {
    const result = shallowRender(
        <Menu
            items={[
                {
                    id: "item1",
                    text: "Item1",
                    to: "url1",
                    exact: true
                },
                {
                    id: "item2",
                    text: "Item2",
                    to: "url2",
                    exact: true,
                    icon: {
                        icon: {
                            id: "iconId"
                        }
                    }
                }
            ]}
        />
    );
    expect(result).toMatchSnapshot();
});
