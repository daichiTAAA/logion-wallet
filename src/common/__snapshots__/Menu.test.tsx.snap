// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Menu"
>
  <MenuItem
    item={
      Object {
        "exact": true,
        "id": "item1",
        "text": "Item1",
        "to": "url1",
      }
    }
  />
  <MenuItem
    item={
      Object {
        "exact": true,
        "icon": Object {
          "icon": Object {
            "id": "iconId",
          },
        },
        "id": "item2",
        "text": "Item2",
        "to": "url2",
      }
    }
  />
</div>
`;
